feat: Add Actor pattern foundation for booking audit logging

hariombalhara opened the pull request:
## What does this PR do?

This PR integrates the existing `BookingAuditService` into the booking lifecycle to log all booking creation, cancellation, and confirmation events. This builds upon the audit logging infrastructure added in the base branch (PR #22817).

**Key Changes:**
- Added `BookingAuditService` calls to `handleNewBooking` for logging booking creation
- Added `BookingAuditService` calls to `handleCancelBooking` for logging cancellations  
- Added `BookingAuditService` calls to `handleConfirmation` for logging confirmations
- Fixed Prisma schema validation error (missing closing brace in `BookingAudit` model)
- Added `Actor` type definition and optional parameter support in booking types

**Important Notes:**
- This PR assumes the `BookingAudit` table and related infrastructure exist from the base branch
- Actor parameter is defined but not fully utilized yet - audit logs currently use `userId` directly
- Audit logging failures are caught and logged as errors (non-blocking)

**Link to Devin run:** https://app.devin.ai/sessions/f3897a417558498991625bf1e8e03fe2  
**Requested by:** @hariombalhara (hariom@cal.com)

## Mandatory Tasks

- [x] I have self-reviewed the code
- [x] I have updated the developer docs if needed (N/A - internal feature)
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works

## How should this be tested?

**Prerequisites:**
1. Ensure the base branch (`feat/booking-audit-log`) migrations have been run to create the `BookingAudit` table
2. Run `npx prisma generate` from `packages/prisma/` to regenerate Prisma client with BookingAudit types

**Test Scenarios:**
1. **Create a booking** - Verify a `BookingAudit` record is created with type `RECORD_CREATED` and action `ACCEPTED`
2. **Cancel a booking** - Verify a `BookingAudit` record is created with type `RECORD_UPDATED` and action `CANCELLED`
3. **Confirm a pending booking** - Verify a `BookingAudit` record is created with type `RECORD_UPDATED` and action `ACCEPTED`

**Database Query to Verify:**
```sql
SELECT * FROM "BookingAudit" WHERE "bookingId" = '<your_booking_id>' ORDER BY "timestamp" DESC;
```

**Known Limitations:**
- May not cover all booking update operations (focus is on main lifecycle events)
- Actor parameter infrastructure exists but is not yet fully integrated from caller side
- Error handling silently logs failures - check server logs if audit records are missing

## Review Checklist

**Critical Items to Review:**
- [ ] Verify BookingAuditService is being called correctly in all three handlers
- [ ] Check if there are other places where bookings are updated/deleted that need audit logging
- [ ] Confirm error handling approach (currently silently fails with log) is acceptable
- [ ] Consider if direct userId usage is sufficient or if Actor object should be fully integrated
- [ ] Verify Prisma schema fix doesn't require a new migration

**Type Safety Concerns:**
- BookingAudit types may show as missing until Prisma client is regenerated
- This is expected and will resolve after running `npx prisma generate`

devin-ai-integration[bot] committed:
feat: Add Actor pattern foundation for booking audit logging

- Define Actor type to represent who performs booking actions (User/System/Attendee)
- Add helper functions for creating and working with actors
- Integrate actor parameter into CancelBookingInput for cancellation tracking
- Create comprehensive documentation for actor pattern implementation
- Foundation for passing actor through all booking services for audit logging

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=24503) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=24503) before we can accept your contribution.<br/>**9** out of **10** committers have signed the CLA.<br/><br/>:white_check_mark: Ryukemeister<br/>:white_check_mark: hbjORbj<br/>:white_check_mark: supalarry<br/>:white_check_mark: Udit-takkar<br/>:white_check_mark: anikdhabal<br/>:white_check_mark: joeauyeung<br/>:white_check_mark: emrysal<br/>:white_check_mark: keithwillcode<br/>:white_check_mark: zhyd1997<br/>:x: iqramjelly<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=24503) it.</sub>

devin-ai-integration[bot] committed:
feat: Add audit logging utility and Actor parameter to booking types

- Create logBookingAudit utility function for centralized audit logging
- Add Actor type import to CreateBookingMeta
- Update CreateBookingMeta to include optional actor parameter
- Actor parameter enables audit logging for who performed booking actions
- All actor parameters are optional for backward compatibility

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>
